File: src/test/java/testdata/java/JsonIgnorePropertiesTestPOJO.java
package testdata.java;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

public class JsonIgnorePropertiesTestPOJO {

    private String username;
    @JsonIgnoreProperties({"users", "aaa", "bbb"})
    private List<Role> roles;

    public class Role {

        private String roleName;
        private List<JsonIgnorePropertiesTestPOJO> users;
    }

}

File: src/test/java/testdata/kotlin/JsonIgnorePropertiesTestPOJO.kt
package testdata.kotlin

class JsonIgnorePropertiesTestPOJO {
    private val username: String? = null

    @com.fasterxml.jackson.annotation.JsonIgnoreProperties("users", "aaa", "bbb")
    private val roles: List<Role>? = null

    class Role {
        private val roleName: String? = null
        private val users: List<JsonIgnorePropertiesTestPOJO>? = null
    }
}